### PR TITLE
Added "How it works?" section to README

### PR DESCRIPTION
**Pull request by dtrenz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,11 +6,21 @@
...
-using enums instead of magic strings that could accidentally be typo-ed.
+`SettingsKit` is a small library & build tool that makes it easier to work
+with app preferences in the iOS Settings app. It also makes working with
+settings a bit safer by using enums instead of "magic" strings, which are
+vulnerable to typos.
 
-![Settings Example](Docs/Screenshots/settings.png)
+
+## How does it work?
+`SettingsKit` comes bundled with a tool that generates an enum (`Settings.swift`)
```
Comment (stkent) [inline README.md:16]: I'd be tempted to say "automatically generates" here to call out extra awesome-sauce (assuming that's accurate!).

Comment (dtrenz) [inline README.md:16]: I'm a fan of awesome-sauce.

Comment (stkent) [inline README.md:16]: It's a bit odd to mention the filename here - only the enum name is important to the consumer, so maybe tweak to

> `SettingsKit` comes bundled with a tool that automatically generates a `Settings` enum...

**Comment by stkent** [comment]
This image sequence would feel more logical to me: enum -> implementation -> result?

**Comment by dtrenz** [comment]
Well, the enum is generated from an existing Settings configuration in the app, aka what is shown in the settings screen example on the far left. Without an existing settings screen configuration (Settings.bundle/Root.plist) nothing can be auto-generated.

So, it goes Settings.bundle -> Settings enum -> Implementation. 

Does that change your mind, or do you still think it would make more sends to change the order?

**Comment by stkent** [comment]
Hmm, based on that info I almost want to say that

Bundle -> Enum -> Implementation -> In-App

makes the most sense, but maybe that's getting out of hand with the number of images?

**Comment by dtrenz** [comment]
Technically, it would be;

`Settings.bundle/Root.plist` -> In-App -> Settings enum -> Implementation

...since you would have a Settings screen already setup and configured before even introducing `SettingsKit` to your project.

I'd be worried that putting the "In-App" screen last might give the false impression that `SettingsKit` auto-generates or auto-configures your `Settings.bundle`for you, which it does not. In fact, it requires that you have already manually set a bundle up, previously.

Also, yeah, 4 screenshots might be pushing it. There are plenty of `Settings.bundle` screenshots below, in the setup section, if that helps?

**Comment by stkent** [comment]
Ahh, I see. I just didn't know enough about iOS settings structure. Now that you've explain, the original sequencing makes sense to me. Thanks!

**Pull request merged**